napari/components/layerlist.py

```diff
@@ -108,70 +117,60 @@ def unselect_all(self, ignore=None):
         ignore : Layer | None
             Layer that should not be unselected if specified.
         """
-        for layer in self:
-            if layer.selected and layer != ignore:
-                layer.selected = False
+        self.selection.intersection_update({ignore} if ignore else {})
 
     def select_all(self):
         """Selects all layers."""
-        for layer in self:
-            if not layer.selected:
-                layer.selected = True
+        self.selection.update(self)
 
     def remove_selected(self):
         """Removes selected items from list."""
-        to_delete = []
-        for i in range(len(self)):
-            if self[i].selected:
-                to_delete.append(i)
-        to_delete.reverse()
-        for i in to_delete:
-            self.pop(i)
-        if len(to_delete) > 0:
-            first_to_delete = to_delete[-1]
-            if first_to_delete == 0 and len(self) > 0:
-                self[0].selected = True
-            elif first_to_delete > 0:
-                self[first_to_delete - 1].selected = True
+        for i in reversed(self):
+            if i in self.selection:
+                self.remove(i)
+
+        # FIXME
```

Comment:
@sofroniewn, can you remind me what the desired behavior is here?  I just want to review it, and _possibly_ not keep it.  simply removing the selected layers is nice and clean, but i can imagine that there's a "next selection step" that someone almost always wants that we were doing here (as we were previously doing).

Reply:
Are you asking what should be selected after removing the selected layers? I'll be honest I don't think there was an intentional decision about this in this implementation (at least not by me). It does look like now something sort of akin to the most top layer above the selection should be selected (though that isn't quite how it is right now empirically).

To be honest I'd be fine with having none selected after this, it sort of makes sense that way - you removed all the selected layers, why have any still selected after that, but i see your point about next most selected too.

Feel free to do what makes most sense to you. I don't think previous behavior was 100% considered